pulp_container/app/registry_api.py

```diff
@@ -349,6 +324,24 @@ def get_dr_push(self, request, path, create=False):
                 raise RepositoryNotFound(name=path)
         return distribution, repository
 
+    def create_dr(self, path, request):
+        try:
+            with transaction.atomic():
+                repository = serializers.ContainerPushRepositorySerializer.get_or_create(
+                    {"name": path}
+                )
+                distribution = serializers.ContainerDistributionSerializer.get_or_create(
+                    {"base_path": path, "name": path}, {"repository": get_url(repository)}
```

Comment:
great, you figured it :+1:

Reply:
here you need to check whether returned distribution has a repo. and if it has it should be same repo-is as on line 330 otherwise we need to raise an error that repo is read-only
how about this workflow:
1. create repo(regular) foo and distribution foo
2. docker push foo